Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -128,6 +146,9 @@ - (void)_update {
     self.updating = YES;
     NSAssert([NSThread isMainThread], @"Update must be called on the main thread");
     NSManagedObjectContext *moc = self.dataStore.viewContext;
+    
+    [self trySavingArticlesToDiskIfNeeded];
```

Comment:
I'd be concerned about adding a second fetch on the main thread here, would be good to profile and/or try to combine with the one below. Ideally it'd be nice to move the fetch below off the main thread as well, could be addressed with 5.8.2 tech debt.